stl/inc/atomic

```diff
@@ -385,20 +388,50 @@ void _Atomic_wait_direct(
         __std_atomic_wait_direct(_Storage_ptr, &_Expected_bytes, sizeof(_Value_type), _Atomic_wait_no_timeout);
     }
 }
+#endif // _HAS_CXX20
 
-template <class _Predicate>
-bool __stdcall _Atomic_wait_indirect_callback(const void* const _Param, void* const _Comparand) noexcept {
-    const auto _Fn = static_cast<const _Predicate*>(_Param);
-    return (*_Fn)(_Comparand);
+#if 1 // TRANSITION, ABI
+inline void _Atomic_lock_spinlock(long& _Spinlock) noexcept {
+    while (_InterlockedExchange(&_Spinlock, 1)) {
+        _YIELD_PROCESSOR();
+    }
 }
 
-template <class _Predicate>
-void _Atomic_wait_indirect(const void* const _Storage, const _Predicate& _Are_equal, void* const _Comparand,
-    unsigned long _Remaining_timeout) noexcept {
-    __std_atomic_wait_indirect(
-        _Storage, _Atomic_wait_indirect_callback<_Predicate>, &_Are_equal, _Comparand, _Remaining_timeout);
+inline void _Atomic_unlock_spinlock(long& _Spinlock) noexcept {
+#if defined(_M_ARM) || defined(_M_ARM64)
+    _Memory_barrier();
+    __iso_volatile_store32(reinterpret_cast<int*>(&_Spinlock), 0);
+    _Memory_barrier();
+#else // ^^^ ARM32/ARM64 hardware / x86/x64 hardware vvv
+    _InterlockedExchange(&_Spinlock, 0);
+#endif // hardware
 }
 
+struct _Spinlock_guard {
+    long& _Spinlock;
+    _Spinlock_guard(long& _Spinlock_) noexcept : _Spinlock(_Spinlock_) {

```

Comment:
I believe that `explicit` is most important for construction from common types like integers and pointers:
```suggestion
    explicit _Spinlock_guard(long& _Spinlock_) noexcept : _Spinlock(_Spinlock_) {
```